### PR TITLE
Add WireGuard

### PR DESCRIPTION
**Pull request by TheZoker**
Add [WireGuard](https://www.wireguard.com/)

> WireGuard® is an extremely simple yet fast and modern VPN that utilizes state-of-the-art cryptography. It aims to be faster, simpler, leaner, and more useful than IPsec, while avoiding the massive headache. It intends to be considerably more performant than OpenVPN. WireGuard is designed as a general purpose VPN for running on embedded interfaces and super computers alike, fit for many different circumstances. Initially released for the Linux kernel, it is now cross-platform (Windows, macOS, BSD, iOS, Android) and widely deployable. It is currently under heavy development, but already it might be regarded as the most secure, easiest to use, and simplest VPN solution in the industry.

### CODE REVIEW

File: bucket/wireguard.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "homepage": "https://www.wireguard.com/",
```
Comment (Ash258) [inline bucket/wireguard.json]:
```suggestion
    "homepage": "https://www.wireguard.com",
```

File: bucket/wireguard.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "description": "WireGuard® is an extremely simple yet fast and modern VPN that utilizes state-of-the-art cryptography.",
+    "version": "0.0.31",
+    "license": "GPLv2",
```
Comment (Ash258) [inline bucket/wireguard.json]:
```suggestion
    "license": "GPL-2.0-or-later",
```

File: bucket/wireguard.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+    "homepage": "https://www.wireguard.com/",
+    "description": "WireGuard® is an extremely simple yet fast and modern VPN that utilizes state-of-the-art cryptography.",
```
Comment (Ash258) [inline bucket/wireguard.json]:
```suggestion
    "description": "Simple and fast VPN client.",
```

File: bucket/wireguard.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    "extract_dir": "WireGuard",
+    "bin": "wireguard.exe",
+    "shortcuts": [
+        [
+            "wireguard.exe",
+            "WireGuard"
+        ]
+    ],
+    "autoupdate": {
```
Comment (Ash258) [inline bucket/wireguard.json]:
```suggestion
    "checkver": {
        "url": "https://www.wireguard.com/install",
        "regex": "distro-status-uptodate\">([\\d.]+)"
    },
    "autoupdate": {
```

**Comment by TheZoker** [comment]
@Ash258 Thanks for the feedback! I implemented all the changes. But I have no clue, why the hashes are wrong 😕

**Comment by linsui** [comment]
This doesn't work because wintun is not installed. See https://github.com/ScoopInstaller/Main/issues/397

File: bucket/wireguard.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{
+    "homepage": "https://www.wireguard.com",
+    "description": "Simple and fast VPN client.",
+    "version": "0.0.31",
+    "license": "GPL-2.0-or-later",
```
Comment (linsui) [inline bucket/wireguard.json]:
```suggestion
    "license": "MIT",
```

> The kernel components are released under the GPLv2, as is the Linux kernel itself. Other projects are licensed under MIT, BSD, Apache 2.0, or GPL, depending on context.

See https://github.com/WireGuard/wireguard-windows

**Comment by TheZoker** [comment]
Hmm that's bad. @Ash258 Do you have any suggestions for this?

**Comment by TheZoker** [comment]
@r15ch13 @Ash258 Can we do something about this?

**Comment by Error1000** [comment]
Could we at least add it to non-portable and install it to Program files and install drivers ( wintun )? I am new here so if I'm missing something please tell me.

**Comment by linsui** [comment]
You can file an issue in the [non-portable bucket](https://github.com/TheRandomLabs/scoop-nonportable).

**Pull request closed without merging**